internal/operator-controller/catalogmetadata/filter/successors_test.go

```diff
@@ -36,6 +36,7 @@ func TestSuccessorsPredicate(t *testing.T) {
 				{
```

Comment:
@joelanford, I think we can cover a few more cases

- Parse a valid semver value with build metadata like 3.0.0-pre+2.beta.1.
- Check if we properly deal with invalid builds values (5.0.0+041).